wallet-new/src/Cardano/Wallet/Kernel.hs

```diff
@@ -119,8 +122,14 @@ handlesOpen mode =
             metadb <- openMetaDB ":memory:"
             migrateMetaDB metadb
             return $ Handles db metadb
-        UseFilePath (DatabasePaths acidDb sqliteDb) -> do
+        UseFilePath (DatabaseOptions acidDb sqliteDb rebuildDB) -> do
+            let deleteMaybe fp = do
+                    when rebuildDB $ do
+                        itsHere <- doesDirectoryExist fp
+                        when itsHere $ removeDirectoryRecursive fp
+            deleteMaybe acidDb
             db <- openLocalStateFrom acidDb defDB
+            deleteMaybe sqliteDb
```

Comment:
I would suggest giving `deleteMaybe` a spin locally to see if it does work correctly with relative vs absolute paths, as I had some surprises before (especially when I was trying to pass things like `../foo/bar` as input).

Hopefully a `ghci` session should give us an answer 😉

Reply:
Seems to work as expected. I also follow what the old wallet did here.